templates/anchor-link.html

```diff
@@ -1 +0,0 @@
-<a class="anchor-link" href="#{{id}}">#</a>
```

Comment:
Do we no longer need this? It appears to have been simply deleted.

Reply:
I think it was never used as a template. Every link seems to have been inlined

Reply:
Actually, I'm not sure that's true, for some reason with this PR the anchor-links in the book are not using the `#` symbol like everywhere else, but this PR doesn't touch that.

Reply:
Alright, found it, it's not super obvious, but when using `insert_anchor_links` inside the zola header thing in the markdown file it will insert that template so it is in fact needed, it's just not used anywhere directly so it's a bit weird. 

@doup I would revert deleting that file and add a simple comment explaining that just so nobody deletes it again in the future.

Reply:
Reverted, and sorry. ^_^U

Btw, I couldn't make Teras white-space removal work with the comments, so the HTML begins just after the comment and I've removed the end-of-line.

Reply:
What's "Teras white-space removal"?

Reply:
I'm refering to this: https://tera.netlify.app/docs/#whitespace-control
And there are users that have make it work: https://github.com/Keats/tera/issues/641

You can add a dash `-` on the closing comment tag and in theory it should remove the line-break between the comment & the link:

```
{# some comment -#}
<a>Link</a>
``` 

I was mentioning this because this piece of HTML gets added on the headers, and the line-break adds a space between the header text and the `#`…

Actually, I'll add line breaks as it'll make the template more readable, and that extra space it's not noticeable…

Hahaha, I sometimes get lost in this small details… 😅